### ISSUE TITLE
Error RARBG: Only 'http' and 'https' schemes are allowed. Parameter name: value

### ISSUE THREAD

**Issue by minitoine** [Jackett/Jackett#5843]
Hi, me again :)

When trying to download magnet from RARBG (download unchecked) to the blackhole, I get this error
` Error: Only 'http' and 'https' schemes are allowed. Parameter name: value`

with the following logs (Don't have anything else, even with enhancing logging sorry :/):

```
Aug 13, 2019 12:23:52 AM | Info | Request finished in 19.2651ms 200 application/json; charset=utf-8
-- | -- | --
Aug 13, 2019 12:23:52 AM | Info | Executed action Jackett.Server.Controllers.BlackholeController.Blackhole (jackett) in 11.6205ms
Aug 13, 2019 12:23:52 AM | Debug | The response will be compressed with 'br'.
Aug 13, 2019 12:23:52 AM | Info | Executing JsonResult, writing value of type 'Newtonsoft.Json.Linq.JObject'.
Aug 13, 2019 12:23:52 AM | Info | Executed action method Jackett.Server.Controllers.BlackholeController.Blackhole (jackett), returned result Microsoft.AspNetCore.Mvc.JsonResult in 3.8483ms.
Aug 13, 2019 12:23:52 AM | Error | Error downloading to blackhole rarbg magnet:?xt=urn:btih:e6c7b14e85128ac74eaa59a928dc534b795125a1&dn=Avengers.Endgame.2019.2160p.BluRay.x265.10bit.SDR.DTS-HD.MA.TrueHD.7.1.Atmos-SWTYBLZ&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce
Aug 13, 2019 12:23:52 AM | Debug | WebClient(HttpWebClient2NetCore).GetString(Url:magnet:?xt=urn:btih:e6c7b14e85128ac74eaa59a928dc534b795125a1&dn=Avengers.Endgame.2019.2160p.BluRay.x265.10bit.SDR.DTS-HD.MA.TrueHD.7.1.Atmos-SWTYBLZ&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce)
Aug 13, 2019 12:23:52 AM | Info | Executing action method Jackett.Server.Controllers.BlackholeController.Blackhole (jackett) with arguments (rarbg, Q2ZESjhNcnVtalc0c0NaRm1Nb0J6VVp0YmZUSEJEaVJQMUFyMXI2YUtmYmNIM1FXWEZTQUZTRUVXVHZ5N0dYSFNCV0JrQlBkYnoyS2N0czhNUE02Z0FZelRsZlVZWFhnNXBOUERoQjFoSUY4QllRcF9nb1gtalNha1o3VlYxTFFhTi1PTm1VSFlWZ005akkwWm1TSzM5OFdJc0lhZVJKVmt6RUJFdWRIQkYyRkRrM2ZqaWR3QmppOGlxNDNpeER3bWVVbDJ2bWdMR0YyWnJQenh5blZkNTl4QTl5OHVJVTZic0QyanFWMDljTGUwaEhNZzJjUFhUN0hHZDZ5bVJaVzM3NlpobGdCMW9DamdPa2lETDlzWW80STNvUjVVTGRpZXlZVWVkbDIwWTgwMVZnVmNlMGw4WEJDV1NkRVJrMGhiSFMyMDNpR1hFbExhMnBKVzl4Q1lRdU1kNFJDVEUxWlZQRzlEQmNraG9PbENxdWZfWnBmQUtRNmZLT0IyN1RfMWhaejFrUzNiOWZfdm5aMTJrdUNnRVl5VmVFel9hMUZhZnJHRHN6SHVoYXdhdFJDSVEzTDBqOFZfWWpUSlpHSFNLRWhCc0E0R1FoWmx1WFZ3bmNLQjVVMUloYWlUNWVnY1NsRWN1OTZFQWc2VnJpN0R3blRwVGVfQllBdVo0TGdjeUZNSGpMNW9mbkwxUGZmRm5PZmZmUHRUbDQ, riomo2p7qgblgknroojoe8irkvfphmb8, Avengers.Endgame.2019.2160p.BluRay.x265.10bit.SDR.DTS-HD.MA.TrueHD.7.1.Atmos-SWTYBLZ) - Validation state: Valid
```

If I check the download torrent, I get a wrong torrent file (Ddos protection css inside).

Edit : saving magnets in the blackhole works through Sonarr. Everything has been working well since few months. It's been few weeks since I have these errors.

**Comment by minitoine** [comment]
Update:
I have the same error using Jackett with mono on Synology, with Jackett on Docker, and Jackett on macOS as a service. Also tried with multiple IPs in Europe and North America.

Do you have the problem on your side?

**Comment by garfield69** [comment]
you have been here before with #5156 
you currently cannot route magnets to the blackhole via the jackett rarbg indexer, it only accepts .torrent (that is, urls that begin with http or https. magnet url begin with magnet, and are not accepted).
I'm not a c# developer so I do not know why that is the case.

this issue here is that rarbg have implemented a clickbait reroute on the url for the .torrent download, where the first click to download takes you to advertising. the second click on the download link starts the .torrent download.
the jackett rarbg indexer would need c# code change (if it is even possible) to cope with this.
setting the indexer to download the magnet at least allows for a reliable download, but not if you are routing to a  blackhole.
that is all I know.

**Comment by minitoine** [comment]
Yeah I know I've been there, but unless I'm wrong or I lost my memory (it is possible!), it was possible to save the magnet link in a .magnet file, right? 
The magnet link is grabbed by Jackett, you have it. Just write it into a file?

**Comment by garfield69** [comment]
Not as far as I am aware. That is the difference between .torrent and magnet. One is a file the other is just a uri. Saving it into a file does not magically turn it into a .torrent and most BitTorrent clients don't read a file containing a magnet string and use it like they could a .torrent

**Comment by minitoine** [comment]
So I guess this is what Sonarr is proposing: the ability to save the magnet uri inside a .magnet file with a custom extension in the black hole.
It might be interesting to add this feature in Jackett as well. Of course, this isn't a torrent file, and most clients don't support that. But at this point, writing a small script to read .magnet files in a folder get the link inside it and send it to a client is not that difficult (this is what I actually do with Sonarr grabs for example.)
Could be just an "expert" Jackett option when a magnet uri is retrieved.

This feature might not be time consuming to develop and is a good workaround for ddos protection.
Otherwise I (and probably others) am completely blocked, can't download the torrent because of cloudfare and save magnet is still not possible.

**Comment by garfield69** [comment]
fix included in jackett pre-release v0.11.640
